File: src/main/java/com/example/spacecourses/Entity/Formation.java
package com.example.spacecourses.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.Date;
import java.util.Set;


@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table( name = "Formation")
public class Formation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer idFormation;
    private String title;
    @Enumerated(EnumType.STRING)
    private Niveau niveau;
    @Temporal (TemporalType.TIMESTAMP)
    private Date start;
    @Temporal (TemporalType.TIMESTAMP)
    private Date end;
    private Integer nbrHeures;
    @Enumerated(EnumType.STRING)
    private Domain domain;

    private Integer nbrMaxParticipant;
    private Integer frais;

    private Integer likes;
    private Integer dislikes;

    @ManyToOne
    @JsonIgnore
    private User formateur;

    @ManyToMany
    @JsonIgnore
    private Set<User> apprenant;

    @OneToMany(mappedBy = "formation")
    @JsonIgnore
    private Set<Quiz> quizzes;


    @OneToMany(mappedBy = "formation")
    @JsonIgnore
    private Set<DatabaseFile> databaseFiles;






}

File: src/app/core/model/Formation.ts
import {Quiz} from "./Quiz";
import {PostComment} from "./PostComment";
import {User} from "./User";
import {Files} from "./files";

export class Formation {

  idFormation!:number;
  title!:string;
  level!:string;
  start!:Date;
  end!:Date;
  nbrHeures!:number;
  domain!:string;
  nbrMaxParticipant!:number;
  frais!:number;
  rating!:number;
  formateur!:User;
  databaseFiles!:Files[];
  postComments:PostComment[];
  apprenant:User[];




}
